### ISSUE TITLE
Error on sync-all --no-refresh --run

### ISSUE THREAD

**Issue by hachi** [TD22057/insteon-mqtt#192]
Hello, I have a quite pleasantly working setup of insteon-mqtt in my home. I'm very happy with the work in this software, but I figure I should report this error for discussion. 

Perhaps the answer is that these are simply mutually exclusive, but I was hoping to run this to prevent spending time refreshing from all devices first.

I would also like to say that I get a very similar error in the case of --no-refresh but without --run as well.

```
sync-all --no-refresh --run
Commanding Modem device 53.c1.da (modem) cmd=sync_all
Syncing 53.c1.da (modem) device
  Adding ID: 1d.9d.c8  grp: 20  type: CTRL  data: 0x14 0x00 0x00:
ERROR: Error running command sync_all on device 53.c1.da (modem)
```

```
2020-02-18 00:56:27 INFO Mqtt: MQTT message insteon/command/modem b'{"cmd": "sync_all", "refresh": false, "dry_run": false, "session": "633491678"}'
2020-02-18 00:56:27 UI Mqtt: Commanding Modem device 53.c1.da (modem) cmd=sync_all
2020-02-18 00:56:27 DEBUG Mqtt: MQTT publish insteon/command/modem/session/633491678 {"type": "MESSAGE", "data": "Commanding Modem device 53.c1.da (modem) cmd=sync_all"} qos=0 ret=False
2020-02-18 00:56:27 DEBUG CommandSeq: Running command 1 of 24
2020-02-18 00:56:27 INFO Modem: Device 53.c1.da (modem) cmd: sync
2020-02-18 00:56:27 UI Modem: Syncing 53.c1.da (modem) device
2020-02-18 00:56:27 DEBUG Mqtt: MQTT publish insteon/command/modem/session/633491678 {"type": "MESSAGE", "data": "Syncing 53.c1.da (modem) device "} qos=0 ret=False
2020-02-18 00:56:27 DEBUG CommandSeq: Running command 1 of 6
2020-02-18 00:56:27 UI Modem:   Adding ID: 1d.9d.c8  grp: 20  type: CTRL  data: 0x14 0x00 0x00:
2020-02-18 00:56:27 DEBUG Mqtt: MQTT publish insteon/command/modem/session/633491678 {"type": "MESSAGE", "data": "  Adding ID: 1d.9d.c8  grp: 20  type: CTRL  data: 0x14 0x00 0x00:"} qos=0 ret=False
2020-02-18 00:56:27 ERROR Mqtt: Error running command sync_all on device 53.c1.da (modem)
Traceback (most recent call last):
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/mqtt/Mqtt.py", line 280, in handle_cmd
    cmd_func(on_done=on_done, **data)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/Modem.py", line 728, in sync_all
    seq.run()
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/CommandSeq.py", line 100, in run
    self.on_done(True, None, None)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/CommandSeq.py", line 133, in on_done
    entry.run(self.protocol, self.on_done)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/CommandSeq.py", line 201, in run
    self.func(*self.args, on_done=on_done, **self.kwargs)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/Modem.py", line 671, in sync
    seq.run()
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/CommandSeq.py", line 100, in run
    self.on_done(True, None, None)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/CommandSeq.py", line 133, in on_done
    entry.run(self.protocol, self.on_done)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/CommandSeq.py", line 201, in run
    self.func(*self.args, on_done=on_done, **self.kwargs)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/Modem.py", line 699, in _sync_add
    on_done=on_done)
TypeError: add_on_device() got multiple values for argument 'on_done'
2020-02-18 00:56:27 DEBUG Mqtt: MQTT publish insteon/command/modem/session/633491678 {"type": "ERROR", "data": "Error running command sync_all on device 53.c1.da (modem)"} qos=0 ret=False
2020-02-18 00:56:27 DEBUG Mqtt: MQTT publish insteon/command/modem/session/633491678 {"type": "END", "data": null} qos=0 ret=False
2020-02-18 00:56:27 DEBUG Mqtt: MQTT writing
```

**Comment by krkeegan** [comment]
Yup, this is my error.

Can you please try adding my repo and pulling from this branch:

https://github.com/krkeegan/insteon-mqtt/tree/Fix_Issue_192

It only has a single commit change that should solve the issue.

**Comment by hachi** [comment]
I'm sorry to say this doesn't seem to cover all problems:

```
2020-02-19 06:00:39 INFO Mqtt: MQTT message insteon/command/modem b'{"cmd": "sync_all", "refresh": false, "dry_run": false, "session": "2119862755"}'
2020-02-19 06:00:39 UI Mqtt: Commanding Modem device 53.c1.da (modem) cmd=sync_all
2020-02-19 06:00:39 DEBUG Mqtt: MQTT publish insteon/command/modem/session/2119862755 {"type": "MESSAGE", "data": "Commanding Modem device 53.c1.da (modem) cmd=sync_all"} qos=0 ret=False
2020-02-19 06:00:39 DEBUG CommandSeq: Running command 1 of 24
2020-02-19 06:00:39 INFO Modem: Device 53.c1.da (modem) cmd: sync
2020-02-19 06:00:39 UI Modem: Syncing 53.c1.da (modem) device
2020-02-19 06:00:39 DEBUG Mqtt: MQTT publish insteon/command/modem/session/2119862755 {"type": "MESSAGE", "data": "Syncing 53.c1.da (modem) device "} qos=0 ret=False
2020-02-19 06:00:39 DEBUG CommandSeq: Running command 1 of 6
2020-02-19 06:00:39 UI Modem:   Adding ID: 1d.9d.c8  grp: 20  type: CTRL  data: 0x14 0x00 0x00:
2020-02-19 06:00:39 DEBUG Mqtt: MQTT publish insteon/command/modem/session/2119862755 {"type": "MESSAGE", "data": "  Adding ID: 1d.9d.c8  grp: 20  type: CTRL  data: 0x14 0x00 0x00:"} qos=0 ret=False
2020-02-19 06:00:39 ERROR Mqtt: Error running command sync_all on device 53.c1.da (modem)
Traceback (most recent call last):
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/mqtt/Mqtt.py", line 280, in handle_cmd
    cmd_func(on_done=on_done, **data)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/Modem.py", line 728, in sync_all
    seq.run()
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/CommandSeq.py", line 100, in run
    self.on_done(True, None, None)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/CommandSeq.py", line 133, in on_done
    entry.run(self.protocol, self.on_done)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/CommandSeq.py", line 201, in run
    self.func(*self.args, on_done=on_done, **self.kwargs)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/Modem.py", line 671, in sync
    seq.run()
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/CommandSeq.py", line 100, in run
    self.on_done(True, None, None)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/CommandSeq.py", line 133, in on_done
    entry.run(self.protocol, self.on_done)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/CommandSeq.py", line 201, in run
    self.func(*self.args, on_done=on_done, **self.kwargs)
  File "/root/insteon-mqtt/venv/lib/python3.7/site-packages/insteon_mqtt/Modem.py", line 699, in _sync_add
    on_done=on_done)
TypeError: add_on_device() got multiple values for argument 'on_done'
2020-02-19 06:00:39 DEBUG Mqtt: MQTT publish insteon/command/modem/session/2119862755 {"type": "ERROR", "data": "Error running command sync_all on device 53.c1.da (modem)"} qos=0 ret=False
2020-02-19 06:00:39 DEBUG Mqtt: MQTT publish insteon/command/modem/session/2119862755 {"type": "END", "data": null} qos=0 ret=False
2020-02-19 06:00:39 DEBUG Mqtt: MQTT writing
```

**Comment by hachi** [comment]
Oh, hang on, have 'install' to test, forgot that step.

**Comment by hachi** [comment]
If you saw them, I deleted my earlier replies. Thank you for this patch.

I've checked and it fixes all existing issues (this and #193 as well) in my environment.

**Comment by krkeegan** [comment]
awesome
